gax/src/test/java/com/google/api/gax/retrying/DirectRetryingExecutorTest.java

```diff
@@ -48,4 +52,32 @@
         new TestResultRetryAlgorithm<String>(apocalypseCountDown, apocalypseException),
         new ExponentialRetryAlgorithm(retrySettings, CurrentMillisClock.getDefaultClock()));
   }
+
+  @Test
+  public void testFutureContainsRetryContext() {
```

Comment:
Can this go to `AbstractRetryingExecutorTest` directly (it looks to be copied in two places `DirectRetryingExecutorTest` and `ScheduledRetryingExecutorTest`, abstract test exists exactly for such cases, to avoid copying)?

Reply:
Will try

Reply:
Actually, I can't do that. The problem is that `getRetryingContext()` is only available on the concrete implementation of `RetryingFuture`, so this test has to existing in each subclass

Reply:
Oh.... This actually indicates, that here should be `RetryingFutureWithContext` pair for `RetryingExeuctorWithContext` =(...

Reply:
I really dont think we should go deeper with this. I'll remove the test as suggested below